lib/index.js

```diff
@@ -161,6 +177,94 @@ class Generator extends EventEmitter {
 
     // Ensure source/destination path, can be configured from subclasses
     this.sourceRoot(path.join(path.dirname(this.resolved), 'templates'));
+
+    // Queues map: generator's queue name => grouped-queue's queue name (custom name)
+    this._queues = {};
+
+    // Add original queues.
+    Generator.queues.forEach(queue => {
+      this._queues[queue] = queue;
+    });
+
+    // Add custom queues
+    if (Array.isArray(this.options.customQueues)) {
+      const customQueues = this.options.customQueues;
+
+      const conflicting = customQueues.find(queue =>
+        Generator.queues.includes(queue.name)
+      );
+      if (conflicting) {
+        throw new Error(
```

Comment:
@SBoudrias maybe we should remove this verification.
Overriding original priority is not recommended, but this will make future proof, since it will not break if we add another original priority.
What do you think?

Reply:
But anyway, I’m not expert, but I think emitting in the constructor will not make difference since listeners aren’t properly registered.

Reply:
Hum, I'm not too sure anymore too 🤦‍♂

I think we can go forward like this for now anyway as it's unlikely to happens - it's not a common runtime error, it's misconfiguration.

If it causes issues during unit tests (yeoman-test), we can address later when we have a clear case.